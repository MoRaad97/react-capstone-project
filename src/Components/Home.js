import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './style/Home.css';
/* tslint:disable: array-callback-return consistent-return  */

const Home = ({ theme = 'default', handleName }) => {
  const [State, setState] = useState({
    filter: 'All',
    search: '',
  });
  const [visible, setVisible] = useState(false);

  const countries = useSelector((state) => state.countries);

  const handleFilter = (e) => {
    const { value } = e.target;
    setState({ ...State, filter: value });
  };

  const handleSearch = (e) => {
    e.preventDefault();
    const { value } = e.target;
    let newValue;
    if (value) {
      newValue = value.trim().toLowerCase();
    } else {
      newValue = '';
    }
    setState(
      { ...State, search: newValue },
    );
  };

  const toggleVisible = () => {
    const scrolled = document.documentElement.scrollTop;
    if (scrolled > 1000) {
      setVisible(true);
    } else if (scrolled <= 1000) {
      setVisible(false);
    }
  };

  window.addEventListener('scroll', toggleVisible);

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth',
    });
  };

  return (
    <section className={theme === 'default' ? 'home home-df' : 'home'}>
      <div className="container">
        <form onSubmit={(e) => e.preventDefault()}>
          <div className="SearchBar-container" onChange={handleSearch} onBlur={handleSearch}>
            <input type="search" placeholder="Search for a country" />
          </div>
          <div className="filter-container">
            <select name="Regions" onChange={handleFilter}>
              <option value="All">Filter by Region</option>
              <option value="Africa">Africa</option>
              <option value="Americas">Americas</option>
              <option value="Asia">Asia</option>
              <option value="Europe">Europe</option>
              <option value="Oceania">Oceania</option>
            </select>
          </div>
        </form>
        <button
          type="button"
          className={theme === 'default' ? 'scroll-up' : 'scroll-up scroll-up-white'}
          style={{ display: visible ? 'inline' : 'none' }}
          onClick={scrollToTop}
          onKeyDown={scrollToTop}
        >
          &#8593;
        </button>
        <div className="cards-container">
          {countries.map((country) => { // eslint-disable-line
            if (State.filter === country.region) {
              if (State.search === '') {
                return (
                  <div
                    className={theme === 'default' ? 'card card-df' : 'card'}
                    key={country.id}
                    id={country.id}
                  >
                    { /* eslint-disable */
                    }
                    <Link to={`${country.name}`}>
                      <img src={country.image} alt={country.name} onClick={handleName} />
                    </Link>
                    <div className="data-container">
                      <h2>{country.name}</h2>
                      <ul>
                        <li>
                          Population:
                          {country.population}
                        </li>
                        <li>
                          Region:
                          {country.region}
                          {' '}

                        </li>
                        <li>
                          Capital:
                          {country.capital}
                        </li>
                      </ul>
                    </div>
                  </div>
                );
              } if (country.name.toLowerCase().includes(State.search)) {
                return (
                  <div
                    className={theme === 'default' ? 'card card-df' : 'card'}
                    key={country.id}
                    id={country.id}
                  >
                    <Link to={`${country.name}`}>
                      <img src={country.image} alt={country.name} onClick={handleName} />
                    </Link>
                    <div className="data-container">
                      <h2>{country.name}</h2>
                      <ul>
                        <li>
                          Population:
                          {country.population}
                        </li>
                        <li>
                          Region:
                          {country.region}
                          {' '}

                        </li>
                        <li>
                          Capital:
                          {country.capital}
                        </li>
                      </ul>
                    </div>

                  </div>
                );
              }
            } else if (State.filter === 'All') {
              if (State.search === '') {
                return (
                  <div
                    className={theme === 'default' ? 'card card-df' : 'card'}
                    key={country.id}
                    id={country.id}
                  >
                    <Link to={`${country.name}`}>
                      <img src={country.image} alt={country.name} onClick={handleName} />
                    </Link>
                    <div className="data-container">
                      <h2>{country.name}</h2>
                      <ul>
                        <li>
                          Population:
                          {country.population}
                        </li>
                        <li>
                          Region:
                          {country.region}
                          {' '}

                        </li>
                        <li>
                          Capital:
                          {country.capital}
                        </li>
                      </ul>
                    </div>

                  </div>
                );
              } if (country.name.toLowerCase().includes(State.search)) {
                return (
                  <div
                    className={theme === 'default' ? 'card card-df' : 'card'}
                    key={country.id}
                    id={country.id}
                  >
                    <Link to={`${country.name}`}>
                      <img src={country.image} alt={country.name} onClick={handleName} />
                    </Link>
                    <div className="data-container">
                      <h2>{country.name}</h2>
                      <ul>
                        <li>
                          Population:
                          {country.population}
                        </li>
                        <li>
                          Region:
                          {country.region}
                          {' '}

                        </li>
                        <li>
                          Capital:
                          {country.capital}
                        </li>
                      </ul>
                    </div>
                  </div>
                );
              }
            }
          })}
        </div>
      </div>
    </section>
  );
};

export default Home;

Home.propTypes = {
  theme: PropTypes.string,
  handleName: PropTypes.func,
};

Home.defaultProps = {
  theme: 'default',
  handleName: undefined,
};
