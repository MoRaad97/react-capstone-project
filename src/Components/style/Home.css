.home {
  background-color: hsl(0, 0%, 98%);
  padding-top: 20px;
  min-height: calc(100vh - 75px);
}

.home-df {
  color: white;
  background-color: #f076a5;
}

.scroll-up {
  width: 45px;
  height: 45px;
  border-radius: 45px;
  background-color: #ec4c8a;
  color: white;
  font-size: 30px;
  position: fixed;
  bottom: 10px;
  right: 10px;
  border: none;
}

.scroll-up-white {
  background-color: rgb(147 146 146);
  color: white;
}

/* search bar */
.SearchBar-container {
  width: 100%;
  margin-bottom: 35px;
}

.SearchBar-container input {
  width: 100%;
  height: 3.5rem;
  font-size: 18px;
  border: none;
  border-radius: 5px;
  text-align: center;
  font-family: "Lato", sans-serif;
}

.SearchBar-container input:focus {
  outline: none;
}

.SearchBar-container input::placeholder {
  text-align: center;
  font-family: "Lato", sans-serif;
}

/* filter */
.filter-container {
  width: 100%;
}

.filter-container select {
  margin-bottom: 30px;
  height: 3.5rem;
  width: 55%;
  border: none;
  border-radius: 5px;
  font-size: 16px;
  text-indent: 15px;
  font-family: "Lato", sans-serif;
}

.filter-container select:focus {
  outline: none;
}

.filter-container option {
  border: none;
  font-family: "Lato", sans-serif;
}

/* cards */
.cards-container {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  row-gap: 30px;
  column-gap: 30px;
}

.card {
  margin: 0 auto 30px;
  width: 85%;
  height: 410px;
  background-color: white;
  box-shadow: 0 0 8px 1px hsl(0deg 0% 85%);
  border-radius: 10px;
}

.card:hover {
  transform: scale(1.02);
  box-shadow: 0 0 10px 5px hsl(0deg 0% 85%);
  transition: 0.5s;
}

.card-df {
  box-shadow: 0 0 8px 1px #6f3d51;
  background-color: #ec4c8a;
  color: white;
}

.card-df:hover {
  transform: scale(1.02);
  box-shadow: 0 0 10px 5px #6f3d51;
  transition: 0.5s;
}

.card img {
  border-radius: 10px 10px 0 0;
  width: 100%;
  height: 165px;
}

.card .data-container {
  padding: 20px 30px;
}

.card h2 {
  margin: 0;
  font-family: "Gill Sans", sans-serif;
  font-weight: 900;
}

.card ul {
  margin: 15px 0 10px;
  padding: 0;
  list-style: none;
  font-family: "Lato", sans-serif;
}

.card li {
  font-size: 17px;
  margin-bottom: 7px;
}
