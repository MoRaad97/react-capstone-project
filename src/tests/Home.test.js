import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import Home from '../Components/Home';

it('Home page Component snapshot test', () => {
  const HomePageRender = TestRenderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(HomePageRender).toMatchSnapshot();
});