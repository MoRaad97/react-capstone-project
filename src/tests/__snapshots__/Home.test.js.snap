// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Component snapshot test 1`] = `
<section
  className="home"
>
  <div
    className="container"
  >
    <form
      action=""
    >
      <div
        className="SearchBar-container"
        onBlur={[Function]}
        onChange={[Function]}
      >
        <input
          placeholder="Search for a country"
          type="search"
        />
      </div>
      <div
        className="filter-container"
      >
        <select
          name="Regions"
          onChange={[Function]}
        >
          <option
            value="All"
          >
            Filter by Region
          </option>
          <option
            value="Africa"
          >
            Africa
          </option>
          <option
            value="Americas"
          >
            Americas
          </option>
          <option
            value="Asia"
          >
            Asia
          </option>
          <option
            value="Europe"
          >
            Europe
          </option>
          <option
            value="Oceania"
          >
            Oceania
          </option>
        </select>
      </div>
    </form>
    <button
      className="scroll-up scroll-up-white"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      â†‘
    </button>
    <div
      className="cards-container"
    />
  </div>
</section>
`;
